cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -285,33 +280,15 @@ class (Ord r) => FiscalRecipient r where
 
 instance FiscalRecipient Int where
 
-prepareProtocolInfo
-  :: forall blk. Consensus.Protocol blk
-  -> TopologyInfo
-  -> ExceptT TxGenError IO (NetworkTopology, ProtocolInfo blk)
-prepareProtocolInfo protocol topologyInfo = do
-  let topologyfile = readTopologyFile (topologyFile topologyInfo)
-  t@(NetworkTopology nodeSetups) <- firstExceptT TopologyFileReadError . newExceptT $ topologyfile
-  nodeId <-
-    case node topologyInfo of
-      RelayId{}  -> left . CurrentlyCannotSendTxToRelayNode $ topologyFile topologyInfo
-      CoreId nid -> return nid
-
-  return ( t
-         , protocolInfo (NumCoreNodes (length nodeSetups))
-                        (CoreNodeId nodeId)
-                        protocol
-         )
```

Comment:
This was mainly there to extract `NumCoreNodes` and `CoreNodeId` from the topology. These values are no longer required to call `protocolInfo`, so I removed this function. I moved L94 to where the topology is actually needed.